components/TextFields/src/private/MDCTextInputCommonFundament.m

```diff
@@ -56,6 +57,27 @@
   return [UIColor lightGrayColor];
 }
 
+@implementation MDCTextInputClearButton
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+  CGFloat width = CGRectGetWidth(self.bounds);
+  CGFloat height = CGRectGetHeight(self.bounds);
+  CGFloat verticalInset = MIN(0, -(MDCTextInputClearButtonTouchTargetSize - height) / 2);
+  CGFloat horizontalInset = MIN(0, -(MDCTextInputClearButtonTouchTargetSize - width) / 2);
+  self.hitAreaInsets =
```

Comment:
Yeah, it is unusual isn't it, but doing it here just ensures that they are always accurate and that the touch target is always the right size.

Reply:
Can we move this point inside code into the text field instead? This would allow us not to rely on the MDCButton hitAreaInsets API which will go away.

Reply:
I had to rebase and force push after #8784, so the original version of this PR is gone, but in it I directly subclassed UIButton and added a hitAreaInsets API to that class. The actual implementation of the button didn't have to change at all after the rebase, I just changed the superclass and removed the UIEdgeInsets property on it since it already got it from MDCButton. I think we could just add it back with zero changes to the implementation file if 
 hitAreaInsets is removed from UIButton before MDCTextFields go away, which I'm pretty sure will happen. I can continue to work on this though if that is not a satisfactory way forward.

Reply:
Ah, I just saw your comment at the bottom @jverkoey. I will try continue to work on this until my grow class at 2pm.

Reply:
I should note that several bugs this week (not ones that I've worked on) have been solved using hitAreaInsets. Maybe we should go through and re-open them!

Reply:
Yes please! Teams are actively expecting us to be moving away from this API.

Reply:
Ah, I've since corrected it but there was a bad typo in https://github.com/material-components/material-components-ios/pull/8790#discussion_r346943568 where I talked about hitAreaInsets being removed from UIButton. I meant removed from MDCButton!